frontend/src/pages/RegistrationsPage.tsx

```diff
@@ -3,110 +3,66 @@ import "./RegistrationsPage.css";
 
 import axios from "axios";
-import {toast, ToastContainer} from "react-toastify";
+import {toast} from "react-toastify";
+import UserRegisterData from "../type/UserRegisterData";
 
 export default function RegistrationsPage() {
 
     const [userEmail, setUserEmail] = useState<string>("");
     const [userName, setUserName] = useState<string>("");
     const [userPassword, setUserPassword] = useState<string>("");
     const [userRepeatPassword, setUserRepeatPassword] = useState<string>("");
-    const [, setEmailError] = useState<string>("");
-    const [, setPasswordError] = useState<string>("");
-    const [, setLoading] = useState<boolean>(true);
 
+    const [errorMessage, setErrorMessage] = useState<string>("");
     const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
 
         event.preventDefault();
 
-
-        if (emailValidation(userEmail) && validatePasswords(userPassword, userRepeatPassword)) {
-            setLoading(true)
-
-            const newUser = {
-                "email": userEmail,
-                "name": userName,
-                "password": userPassword
-            }
-
-            axios.post("/auth/register", newUser)
-                .then((reponse) => reponse.data)
-                .then((data) => data)
-                .catch(error => {
-                    if (error.response?.message !== undefined) {
-                        toast(error.response.message);
-                        console.log(error)
-                    }
-                })
-                .finally(() => setLoading(false));
-
-        }
-
-
-    }
-
-    const emailValidation = (userMail: string) => {
-        if (!/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(userMail)) {
-            setEmailError("Email not valid")
-            return false
-        }
-        setEmailError("");
-        return true;
-    }
-
-    const validatePasswords = (password: string, repeatPassword: string) => {
-        if (password.length < 6) {
-            setPasswordError("Password too short")
-            return false
-        }
-        if (password.length > 100) {
-            setPasswordError("Password too long")
-            return false
-        }
-        if (!/[a-z]/.test(password)) {
-            setPasswordError("lowercase letter required")
-            return false
-        }
-        if (!/[A-Z]/.test(password)) {
-            setPasswordError("uppercase letter required")
-            return false
-        }
-        if (!/\d/.test(password)) {
-            setPasswordError("number required")
-            return false
-        }
-        if (!/[ `!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/.test(password)) {
-            setPasswordError("special character required")
-            return false
-        }
-        if (password !== repeatPassword) {
-            setPasswordError("Passwords do not match")
-            return false
```

Comment:
Ich würde die frontend validierung schon drin lassen, dann kannst du direkt und schnell dem nutzer anzeigen was noch nicht passt. Vielleicht lagerst du die Logik dafür in eine Service funktion aus? 🤔

Reply:
ich nehme erst einmal im Projekt auf. erarbeite das erst später